internal/webhooks/inclusterippool.go

```diff
@@ -147,3 +156,50 @@ func (webhook *InClusterIPPool) validate(_, newPool *v1alpha1.InClusterIPPool) (
 
 	return //nolint:nakedret
 }
+
+func validateAddresses(newPool *v1alpha1.InClusterIPPool) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if newPool.Spec.Subnet != "" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "subnet"), newPool.Spec.Subnet, "subnet may not be used with addresses"))
+	}
+
+	if newPool.Spec.First != "" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "start"), newPool.Spec.First, "start may not be used with addresses"))
+	}
+
+	if newPool.Spec.Last != "" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "end"), newPool.Spec.Last, "end may not be used with addresses"))
+	}
+
+	if newPool.Spec.Gateway == "" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "gateway"), newPool.Spec.Gateway, "gateway is required when using addresses"))
+	}
+
+	if newPool.Spec.Prefix == 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "prefix"), newPool.Spec.Prefix, "a valid prefix is required when using addresses"))
+	}
+
+	ips := make([]netaddr.IP, len(newPool.Spec.Addresses))
+	for i, address := range newPool.Spec.Addresses {
+		ip, err := netaddr.ParseIP(address)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "addresses"), address, "provided address is not a valid ip address"))
+			continue
+		}
+		ips[i] = ip
+	}
+
+	prefix, err := netaddr.ParseIPPrefix(fmt.Sprintf("%s/%d", newPool.Spec.Addresses[0], newPool.Spec.Prefix))
```

Comment:
Subnet could be set to this prefix during defaulting.

Reply:
We've responded to this in the previous comment.